persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickDbProvider.scala

```diff
@@ -77,7 +91,14 @@ private[lagom] object SlickDbProvider {
 
     lifecycle.addStopHook { () =>
       namingContext.unbind(jndiDbName)
-      slickDb.shutdown
+      Future.successful(Done)
+    }
+
+    coordinatedShutdown.addTask(
+      CoordinatedShutdown.PhaseBeforeActorSystemTerminate,
+      s"shutdown-managed-read-side-slick"
+    ) { () =>
+      slickDb.shutdown.map(_ => Done)
     }
```

Comment:
This is the actual fix for Slick. Instead of shutting down the DB early (on `lifecycle.addStopHook`) just unbind it from JNDI's `namingContext` and use a task on a late phase of `coordinatedShutdown` instead.